src/Microsoft.DotNet.Build.Tasks/PackageFiles/crossgen.sh

```diff
@@ -54,45 +49,46 @@ crossgen_single()
 {
     __file=$1
     if [[ $__file != *.ni.dll && $__file != *.ni.exe ]]; then
-        if [[ ($__file == *.dll && -e ${__file/.dll/.ni.dll}) || ($__file == *.exe && -e ${__file/.exe/.ni.exe}) ]]; then
-            echo "$__file has already been crossgen'd.  Skipping."
+        set +e
+        $__crossgen /Platform_Assemblies_Paths $__sharedFxDir:$__toolsDir /JitPath $__sharedFxDir/libclrjit.$__libraryExtension /nologo /MissingDependenciesOK /ReadyToRun $__file > /dev/null
+        if [ $? -eq 0 ]; then
+            __outname="${__file/.dll/.ni.dll}"
+            __outname="${__outname/.exe/.ni.exe}"
+            echo "$__file -> $__outname"
         else
-            $__crossgen /Platform_Assemblies_Paths $__sharedFxDir:$__toolsDir /nologo /MissingDependenciesOK /ReadyToRun $__file > /dev/null
-            if [ $? -eq 0 ]; then
-                __outname="${__file/.dll/.ni.dll}"
-                __outname="${__outname/.exe/.ni.exe}"
-                echo "$__file -> $__outname"
-            else
-                echo "Unable to successfully compile $__file"
-            fi
+            echo "Unable to successfully compile $__file"
         fi
+        set -e
     fi
 }
 
-if [ ! -z $BUILDTOOLS_SKIP_CROSSGEN ]; then
+if [ ! -z ${BUILDTOOLS_SKIP_CROSSGEN:-} ]; then
     echo "BUILDTOOLS_SKIP_CROSSGEN is set. Skipping crossgen step."
     exit 0
 fi
 
-if [[ -z "$1" || "$1" == "-?" || "$1" == "--help" || "$1" == "-h" ]]; then
+if [[ -z "${1:-}" || "$1" == "-?" || "$1" == "--help" || "$1" == "-h" ]]; then
     usage
 fi
 
+__MyGetFeed=${BUILDTOOLS_CROSSGEN_FEED:-https://dotnet.myget.org/F/dotnet-core/api/v3/index.json}
 __targetDir=$1
 __scriptpath=$(cd "$(dirname "$0")"; pwd -P)
 __toolsDir=$__scriptpath/../Tools
 __dotnet=$__toolsDir/dotnetcli/dotnet
-__packagesDir="${NUGET_PACKAGES:-${__scriptpath}/../packages}"
+__packagesDir=${NUGET_PACKAGE:-$__scriptpath/../packages}
```

Comment:
I will undo this one change (i was using a version of init-tools.sh that didn't have and added it back but slightly differently)!